modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/AbstractIndexScan.java

```diff
@@ -136,18 +144,24 @@ public RelOptCost computeSelfCost(RelOptPlanner planner, RelMetadataQuery mq) {
 
             cost = 0;
 
-            if (lowerCondition() != null) {
-                double selectivity0 = mq.getSelectivity(this, RexUtil.composeConjunction(builder, lowerCondition()));
+            if (type == Type.HASH && lowerBound() != null) {
```

Comment:
could you please explain why lowerBound is importannt here?

Reply:
have added the comment to the code. Is it helpful?

Reply:
yes, now it is clear. Thanks